sp/src/game/server/ez2/npc_pitdrone.cpp

```diff
@@ -210,6 +229,57 @@ int CNPC_PitDrone::RangeAttack1Conditions( float flDot, float flDist )
 	return(BaseClass::RangeAttack1Conditions( flDot, flDist ));
 }
 
+//=========================================================
+// Start task - selects the correct activity and performs
+// any necessary calculations to start the next task on the
+// schedule. 
+//
+// Overridden for bullsquids to play specific activities
+//=========================================================
+void CNPC_PitDrone::StartTask( const Task_t *pTask )
+{
+	CGravityVortexController *pVortex;
+
+	switch (pTask->iTask)
+	{
+	case TASK_PREDATOR_SPAWN:
+		// TODO - This should probably have an animation
+
+		// Play a sound
+		EmitSound( "NPC_PitDrone.SummonBackup" );
+
+		// Create a portal
+		pVortex = CGravityVortexController::Create( GetAbsOrigin(), 0.0f, 0.0f, 0.0f, this );
+		pVortex->AddContext( "owner_classname:npc_pitdrone" );
+
+		// If this mate is my squadmate, break off into a new squad for now
+		if (this->GetSquad() == NULL)
+		{
+			// Autogenerate a name for the new squad with GetDebugName()-entindex(). It's important that it has a name
+			g_AI_SquadManager.CreateSquad( AllocPooledString(UTIL_VarArgs("%s-%i", GetDebugName(), entindex() )))->AddToSquad( this );
+		}
+
+		// If we are in a squad and that squad has a name, apply that squad's name to the Xen singularity
+		if (GetSquad() && !Matcher_Match( GetSquad()->GetName(), ""))
```

Comment:
The first character of an empty string will be its null terminator. To check for that, you would usually do something like this, which I would recommend using instead:
```suggestion
		if (GetSquad() && GetSquad()->GetName()[0] != '\0')
```
*(Note that the `#include "mapbase/matchers.h"` at the top might no longer be needed with this change)*

Reply:
Thanks, I think this is what I was looking for. That is very ugly though. As just a thought, maybe we could make that a utility function?

Another question - are we assured that the string won't be NULL in this case? I don't want this to throw an NPE. I could add a null check beforehand, but if we're guaranteed this value is non-null then it's moot.

Reply:
How about this instead?
```suggestion
		if (GetSquad() && GetSquad()->GetName() && GetSquad()->GetName()[0])
```

Reply:
That looks good - don't we need to compare that first char against '\0' though?

Reply:
'\0' is equivalent to `0`, so this should be fine. I initially wrote it out explicitly due to personal preference. Other code in the SDK tends to alternate.